--- ode/src/Makefile.am	Fri Jul 11 08:16:50 2008
+++ ode/src/Makefile.am	Wed Aug  8 08:43:51 2012
@@ -8,7 +8,13 @@
 noinst_LTLIBRARIES = libfast.la
 libfast_la_SOURCES = fastldlt.c fastltsolve.c fastdot.c fastlsolve.c
 
-
+# to create child classes of dJoint, we need to make more of
+# the internal data structures accessible.
+libode_la_includedir = $(includedir)/ode-internals
+libode_la_include_HEADERS = objects.h \
+			array.h \
+			obstack.h \
+			config.h
 
 lib_LTLIBRARIES = libode.la
 
--- ode/src/joints/Makefile.am	Wed Jun  4 23:47:13 2008
+++ ode/ode/src/joints/Makefile.am	Wed Aug  8 08:43:51 2012
@@ -4,6 +4,9 @@
 
 noinst_LTLIBRARIES = libjoints.la
 
+libjoints_la_includedir = $(includedir)/ode/joints
+libjoints_la_include_HEADERS = joint.h
+
 libjoints_la_SOURCES =  joints.h \
                         joint.h joint.cpp \
                         joint_internal.h \
 
